core/src/main/java/hudson/cli/OnlineNodeCommand.java

```diff
@@ -40,29 +45,61 @@
 @Extension
 public class OnlineNodeCommand extends CLICommand {
 
-    @Argument(metaVar="NAME", usage="Agent name, or empty string for master")
-    public String computerName;
+    @Argument(metaVar = "NAME", usage = "Agent name, or empty string for master", required = true, multiValued = true)
+    private List<String> nodes;
 
     @Override
     public String getShortDescription() {
-
         return Messages.OnlineNodeCommand_ShortDescription();
     }
 
     @Override
     protected int run() throws Exception {
-
+        boolean errorOccurred = false;
         final Jenkins jenkins = Jenkins.getActiveInstance();
 
-        Computer computer = jenkins.getComputer(computerName);
+        final HashSet<String> hs = new HashSet<String>();
+        hs.addAll(nodes);
+
+        List<String> names = null;
+
+        for (String node_s : hs) {
+            Computer computer = null;
+
+            try {
+                computer = jenkins.getComputer(node_s);
 
-        if (computer == null) {
-            throw new IllegalArgumentException(hudson.model.Messages.Computer_NoSuchSlaveExists(computerName, null));
-        } else {
-            computer.cliOnline();
+                if (computer == null) {
+                    if (names == null) {
+                        names = new ArrayList<String>();
```

Comment:
I don't agree here. The result is used only if user makes a mistake (the requested computer isn't found) - computation of that is a bit time consuming operation. We should not to do the calculation for every CLI call because it isn't necessary.

When we compute it - we need to collect all computers into the list and to calculate the nearest one. In that case we can defend the overhead by trying to prepare proper answer - we need to search through the whole inventory:-).